pkg/util/util_test.go

```diff
@@ -166,6 +166,57 @@ func TestMergeResourceRequests(t *testing.T) {
 	assert.Equal(t, *resource.NewQuantity(123, resource.DecimalSI), merged.Resources.Requests[corev1.ResourceRequestsEphemeralStorage])
 }
 
+func TestAffinityDefault(t *testing.T) {
+	generalSpec := v1.JaegerCommonSpec{}
+	specificSpec := v1.JaegerCommonSpec{}
+
+	merged := Merge([]v1.JaegerCommonSpec{specificSpec, generalSpec})
+
+	assert.Nil(t, merged.Affinity)
+}
+
+func TestAffinityOverride(t *testing.T) {
+	generalSpec := v1.JaegerCommonSpec{
+		Affinity: &corev1.Affinity{
+			NodeAffinity: &corev1.NodeAffinity{},
+		},
+	}
+	specificSpec := v1.JaegerCommonSpec{
+		Affinity: &corev1.Affinity{
+			PodAffinity: &corev1.PodAffinity{},
+		},
+	}
+
+	merged := Merge([]v1.JaegerCommonSpec{specificSpec, generalSpec})
+
+	assert.NotNil(t, merged.Affinity)
+	assert.NotNil(t, merged.Affinity.PodAffinity)
+	assert.Nil(t, merged.Affinity.NodeAffinity)
+}
+
+func TestMergeTolerations(t *testing.T) {
+	generalSpec := v1.JaegerCommonSpec{
+		Tolerations: []corev1.Toleration{{
+			Key: "toleration1",
+		}},
+	}
+	specificSpec := v1.JaegerCommonSpec{
+		Tolerations: []corev1.Toleration{{
+			Key: "toleration1",
+		}, {
+			Key: "toleration2",
+		}},
+	}
+
+	merged := Merge([]v1.JaegerCommonSpec{specificSpec, generalSpec})
+
+	// Keys are not unique, so should be aggregation of all tolerations
+	assert.Len(t, merged.Tolerations, 3)
+	assert.Equal(t, "toleration1", merged.Tolerations[0].Key)
```

Comment:
This *looks* wrong at a first glance, but reading the docs makes it clear that it's OK to have two entries with the same key/operator/value, as long as the effect is different. Perhaps add a link to the relevant docs, to avoid future confusion?

Reply:
Done